res/enkf/local_ministep.py

```diff
@@ -1,84 +1,144 @@
     def numActiveData(self):
-        return self._data_size()
+        return len(self.parameters) + len(self.row_scaling_parameters)
 
+    @deprecated("Will be removed")
     def addNode(self, node):
-        _lib.local.ministep.add_obsdata_node(self, node)
+        raise NotImplementedError("Adding node to ministep is not supported")
 
+    @deprecated(
+        "Will be removed, use: add_observation",
+        removed_in="2.37",
+        current_version=__version__,
+    )
     def attachObsset(self, obs_set):
         assert isinstance(obs_set, LocalObsdata)
-        _lib.local.ministep.attach_obsdata(self, obs_set)
-
-    def row_scaling(self, key) -> RowScaling:
-        if not self._has_active_data(key):
-            raise KeyError(f"Unknown key: {key}")
-
-        return _lib.local.ministep.get_or_create_row_scaling(self, key)
+        for node in obs_set:
+            self.observations[node.key()] = Observation(
+                node.key(), node.getActiveList().get_active_index_list()
+            )
+
+    @deprecated(
+        "Will be removed, use: add_row_scaling and get the row_scaling property",
+        removed_in="2.37",
+        current_version=__version__,
+    )
+    def get_or_create_row_scaling(self, name):
+        return self.row_scaling(name)
 
+    @deprecated(
+        "Will be removed, use: add_row_scaling",
+        removed_in="2.37",
+        current_version=__version__,
+    )
+    def row_scaling(self, parameter_name) -> RowScaling:
+        if parameter_name in self.row_scaling_parameters:
+            return self.row_scaling_parameters[parameter_name].row_scaling
+        elif parameter_name in self.parameters:
+            parameter = self.parameters[parameter_name]
+            row_scaling = RowScaling()
+            self.row_scaling_parameters[parameter_name] = RowScalingParameter(
+                parameter.name, row_scaling, parameter.index_list
+            )
+            del self.parameters[parameter_name]
+            return row_scaling
+        raise KeyError(f"No such parameter: {parameter_name}")
+
+    @deprecated(
+        "Use property: observations", removed_in="2.37", current_version=__version__
+    )
     def getLocalObsData(self):
         """@rtype: LocalObsdata"""
-        return _lib.local.ministep.get_obsdata(self)
+        return self.observations.values()
 
+    @deprecated(
+        "Will be turned into property: name",
+        removed_in="2.37",
+        current_version=__version__,
+    )
     def name(self):
-        return self._name()
+        return self._name
 
+    @deprecated(
+        "Will be removed, and turned into property: name",
+        removed_in="2.37",
+        current_version=__version__,
+    )
     def getName(self):
         """@rtype: str"""
         return self.name()
 
-    def free(self):
-        self._free()
+    def observation_config(self):
+        return [observation.to_tuple() for observation in self.observations.values()]
```

Comment:
could this not be 
```suggestion
        return [(key, value) for observation in self.observations.items()]
```
or could the key in the dict and the name of observation differ?

Reply:
This also removes the need of `to_tuple()` function